Client/mods/deathmatch/logic/luadefs/CLuaResourceDefs.cpp

```diff
@@ -337,28 +337,16 @@ int CLuaResourceDefs::GetResourceGUIElement(lua_State* luaVM)
     return 1;
 }
 
-int CLuaResourceDefs::GetResourceDynamicElementRoot(lua_State* luaVM)
+CClientEntity* CLuaResourceDefs::GetResourceDynamicElementRoot(lua_State* luaVM, std::optional<CResource*> resource)
 {
-    CResource*       pResource = NULL;
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pResource);
-
-    if (!argStream.HasErrors())
+    CResource* pResource = resource.value_or(&lua_getownerresource(luaVM));
+    if (!pResource->IsActive())
```

Comment:
This check caused problems with a previous pull request, I don't think we should use it here either

Reply:
Thats the original version ported to ArgumentParser

Reply:
The client-sided equivalent does not include this check, so you should only remove it from the client-sided

Reply:
I added this check as per previous review

Reply:
I think you should do some tests then, just use the map editor for a few minutes, because there was a problem with the previous pull request, and if it doesn't cause any problems, close the conversation and leave it there

Reply:
Works fine for me. I didnt encounter any issues